### PR TITLE
Add `CAST()`

### LINKED ISSUE #11: Resolve TIMESTAMP() vs CAST()
Following on from discussions at the [October 2019 IVOA interop meeting](https://wiki.ivoa.net/twiki/bin/view/IVOA/InterOpOct2019), a number of people raised comments and issues about TIMESTAMP() and CAST().

This is a placholder issue that represents an ongoing discussion on the [DAL mailing list](http://mail.ivoa.net/pipermail/dal/). The discussion will continue as normal on the mailing list, and this GitHub issue will be closed when the discussion reaches a conclusion.

### PR DESCRIPTION
**Pull request by gmantele**
This PullRequest is a proposal for adding the function `CAST()`.
In addition of adding basic cast operations, it also aims to create `TIMESTAMP` values.

The section '3. Type System' has also been updated:

- `DOUBLE` -> `DOUBLE PRECISION`
- `INTERVAL` description clearly states that ADQL does not provide any way to either create or manipulate intervals. This will be done in a next ADQL version. However, service providers can implement UDFs to do so.
- Fix the explanation about how to create `TIMESTAMP` literals: with `CAST()` instead of `TIMESTAMP()` (which no longer exists)

About `BIT`, `INTERVAL`, `BINARY`, `VARBINARY`, `CLOB` and `BLOB` which can not be created (even with `CAST()`) nor manipulated currently in ADQL, I chose to keep them as already defined in section '3. Type System'. This section aims to link ADQL to the TAP metadata. Even if a datatype is not supported (for creation or manipulation) by default in ADQL, service providers should be allowed to publish columns with those types. They could be useful to some services. Besides, even if not natively supported in ADQL, implementations could choose to do something with them by defining UDFs. These UDFs can then be a very good base (because dedicated to concrete use-cases) for future versions of ADQL.

Fixes #11

### CODE REVIEW

**Comment by msdemlei** [comment]
I probably wouldn't have bothered to move the reserved words between between
SQL_reserved_words and ADQL_reserved_words, and for NULL at least, I'd mildly
advocate moving it back, but since it has no operational consequences I don't
really mind.

Other than that, this looks really good to me.  I particularly like the
table of required conversions.

**Review by molinaro-m — APPROVED** [review]
Considering the foreground this PR sits on and Markus's review I add here my approval. I leave it to the editor to decide whether a second review might be needed (from a more knowledgeable person).

**Comment by gmantele** [comment]
I leave it open a couple of days before merging it so that giving a bit more time to anybody else to review it if he/she feels this need (I mainly think to @Zarquan who spent some time writing most of the current material about the type system and CAST, and to @pdowler who might see a link with DALI).

**Review by Zarquan — APPROVED** [review]
Yep, all looks OK.

**Pull request merged**
